src/libaktualizr/uptane/tuf.h

```diff
@@ -80,36 +80,47 @@ class TimeStamp {
 
 std::ostream &operator<<(std::ostream &os, const TimeStamp &t);
 
-class HardwareIdentifier {
+class BaseId {
  public:
   // https://github.com/advancedtelematic/ota-tuf/blob/master/libtuf/src/main/scala/com/advancedtelematic/libtuf/data/TufDataType.scala
   static const int kMinLength = 0;
   static const int kMaxLength = 200;
 
-  static HardwareIdentifier Unknown() { return HardwareIdentifier("Unknown"); }
-  explicit HardwareIdentifier(const std::string &hwid) : hwid_(hwid) {
+  explicit BaseId(const std::string &id) : id_(id) {
     /* if (hwid.length() < kMinLength) {
       throw std::out_of_range("Hardware Identifier too short");
     } */
-    if (kMaxLength < hwid.length()) {
+    if (kMaxLength < id.length()) {
       throw std::out_of_range("Hardware Identifier too long");
     }
   }
 
-  std::string ToString() const { return hwid_; }
+  const std::string &ToString() const { return id_; }
```

Comment:
This feels like a dangerous optimisation. If the caller keeps the std::string reference alive longer than the `BaseId` object, then they will reference an invalid pointer. Given that it is hard to statically validate these things, I think we should pray for a Sufficiently Intelligent Compiler, i.e. accept the extra cost.